frontend/components/admin/pages/AdvertisementEditor.tsx

```diff
@@ -0,0 +1,347 @@
+'use client'
+import React, { useState, FormEvent } from 'react'
+
+import {
+  Box,
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Stack,
+  Link,
+  useToast,
+  HStack,
+  Text,
+} from '@chakra-ui/react'
+import { THEME_COLOR } from '../../../constants'
+
+import { AdvertisementType } from '../../../types/AdvertisementType'
+import { advertisementSchema } from '../../../zod'
+import { ZodIssue } from 'zod'
+
+type AdvertisementEditorProps = {
+  advertisement: AdvertisementType
+}
+export function AdverrisementEditor({
+  advertisement,
+}: AdvertisementEditorProps) {
+  const selectedAdvertisement: AdvertisementType = {
+    id: advertisement.id,
+    name: advertisement.name,
+    url: advertisement.url,
+    imageUrl: advertisement.imageUrl,
+    author: advertisement.author,
+    startFrom: new Date(advertisement.startFrom),
+    endAt: new Date(advertisement.endAt),
+  }
+  const toast = useToast()
+  const [advertisementData, setAdvertisementData] = useState<AdvertisementType>(
+    selectedAdvertisement,
+  )
+  const [errors, setErrors] = useState<ZodIssue[]>([])
+  const [isSubmitting, SetIsSubmitting] = useState(false)
+  const updateAdvertisement = async (event: FormEvent) => {
+    try {
+      event.preventDefault()
+
+      const formData = {
+        name: advertisementData.name,
+        url: advertisementData.url,
+        author: advertisementData.author,
+        imageUrl: advertisementData.imageUrl,
+        startFrom: advertisementData.startFrom.toISOString(),
+        endAt: advertisementData.endAt.toISOString(),
+      }
+
+      const advertisementValidationResult =
+        advertisementSchema.safeParse(formData)
+      if (advertisementValidationResult.success === false) {
+        setErrors(advertisementValidationResult.error.issues)
+        toast({
+          title: 'フォームの入力に誤りがあります',
+          status: 'error',
+          position: 'top',
+          duration: 3000,
+        })
+        return
+      }
+
+      event.preventDefault()
+
+      SetIsSubmitting(true)
+
+      const response = await fetch(
+        `${process.env.NEXT_PUBLIC_BACKEND_URL}/advertisement`,
+        {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            id: advertisementData.id,
+            name: advertisementData.name,
+            url: advertisementData.url,
+            imageUrl: advertisementData.imageUrl,
+            author: advertisementData.author,
+            startFrom: advertisementData.startFrom,
+            endAt: advertisementData.endAt,
```

Comment:
```suggestion
            id: advertisementData.id,
            name: advertisementData.name,
            url,
            imageUrl,
            author,
            startFrom,
            endAt
```

上のコメントの対応すれば省略できますね！